src/lib/utils/ct_utils.h

```diff
@@ -324,6 +331,30 @@ inline CT::Mask<T> all_zeros(const T elem[], size_t len) {
    return CT::Mask<T>::is_zero(sum);
 }
 
+/**
+* Compare two arrays of equal size and return a Mask indicating if
+* they are equal or not. The mask is set if they are identical.
+*/
+template <typename T>
+inline CT::Mask<T> is_equal(const T x[], const T y[], size_t len) {
+   volatile T difference = 0;
+
+   for(size_t i = 0; i != len; ++i) {
+      difference = difference | (x[i] ^ y[i]);
+   }
+
+   return CT::Mask<T>::is_zero(difference);
+}
```

Comment:
This could be based on ranges, no? Something along these lines, with `CT::Mask<>` fairy-dust sprinkled in:

```suggestion
template <std::ranges::contiguous_range X, std::ranges::contiguous_range Y>
    requires std::same_as<std::ranges::range_value_t<X>, std::ranges::range_value_t<Y>>
constexpr auto is_equal(X&& x, Y&& y)
{
    volatile auto difference = static_cast<std::ranges::range_value_t<X>>(std::ranges::size(x) != std::ranges::size(y));

    auto ix = std::ranges::begin(x);
    auto iy = std::ranges::begin(y);

    // TODO C++23: std::views::zip(x, y)
    //             future-me, please make sure that zip() is constant-time
    while(ix != std::ranges::end(x) && iy != std::ranges::end(y))
    {
        difference |= *ix ^ *iy;
        ++ix;
        ++iy;
    }

    return difference;
}```

Reply:
Maybe so. But tbh for this kind of code I like writing things in the most brain-dead way possible because it's usually (fairly) obvious what the compiler is going to do with it.

Some span based overloads here might be nice though. As with #3757 the goal here is primarily to remove usage of functions in `mem_ops.h` especially when they were already redundant with other functionality.